snuba/datasets/configuration/generic_metrics/entities/sets.yaml

```diff
@@ -101,3 +101,16 @@ validators:
       required_filter_columns: ["org_id", "project_id"]
 required_time_column: timestamp
 partition_key_column_name: org_id
+entity_subscription:
+  validators:
+    -
+      validator: AggregationValidator
+      args:
+        max_allowed_aggregations: 3
+        disallowed_aggregations: [having, orderby]
+  processors:
+    -
+      processor: AddColumnCondition
+      args:
+        extra_condition_data_key: organization
+        extra_condition_column: org_id
```

Comment:
There is some implicit semantic validation done when the system tries to build the entity. For entity subscription processors and validators, this happens [here](https://github.com/getsentry/snuba/pull/3384/files#diff-530c557e763d4aa8baeb0c82eb931da0d8e7206d886d3b266357330ed9506145R100). The builder looks up the class name by a base class config. As a result, there won't be a misconfiguration when compiling it. However, right now, we have no cues to inform users about these types of misconfiguration before runtime. The `make validate-configs` command only checks that the schema is congruent to json_schema definition (no semantic validation).

Reply:
If entity validations are processors are separate as you say (which I am not sure about b/c of https://github.com/getsentry/snuba/pull/3384#discussion_r1025863707), I don't really think you need the entity subscription parent at all.

Why not just have:
```
subscription_validators:
  - validator
subscription_processors:
  - processor

The entity subscription parent and the entity prefix serves no value anymore since we are already within the context of an entity

Reply:
This makes sense to me. Before making any changes, I'd like to point out a few implications with this change:
* `SubscriptionData` class has an attribute `entity_subscription: EntitySubscription` as of today. This change would require us to change that interface.
* New use cases might be added to entity_subscriptions as we move towards providing service owners with more autonomy. This would require new attributes to be added to this class.

Reply:
What do you imagine these new use cases for entity subscriptions be? A processor already allows a user quite a lot of autonomy to transform a query in any way.

Reply:
I don't think I know of any real use-cases today and I'm unsure about the future direction of subscriptions. Maybe adding rate limiting functionality to subscriptions.. idk? If most speculative use cases can fit within processors and validators, then we don't need to design for features that don't exist.

Reply:
Updated to split subscription_validators and subscription_processors independently in Entity class